pytorch_lightning/strategies/parallel.py

```diff
@@ -72,6 +72,14 @@ def world_size(self) -> int:
     def is_global_zero(self) -> bool:
         return self.global_rank == 0
 
+    @property
+    def parallel_devices(self):
+        return self._parallel_devices
+
+    @parallel_devices.setter
+    def parallel_devices(self, parallel_devices):
+        self._parallel_devices = parallel_devices
```

Comment:
here we see a hint at the difference between the proposals #11097 and #7650. 

If a setter like this one gets introduced, and other values depend on this value here, these would have to be re-computed everytime this setter is invoked (depending on the requirements of the plugin, error handling would be needed to). If multiple of these setters are necessary, things get complicated and logic gets duplicated. #7650 would solve this, because it would set all attributes in one shot together. I haven't found the time to investigate in this direction further.

Reply:
+1 for #7650. The accelerator_connector should be the only place allowed to set/modify these parameters (other than to pass in when creating the object). With #7650 proposal, we can call the lazy_init at the end of the accelerator_connector __init__, which will be way less error-prone and the behavior for properties will be more consistent and robust

This PR take the  #11097 easy way for now to unblock accelerator_connector rewrite..